src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -112,6 +136,25 @@ def _rbd_image(self, ioctx, pool_name, image_name):
                 })
             stat['snapshots'].append(snap)
 
+        # disk usage
+        if 'fast-diff' in stat['features_name']:
```

Comment:
Nit: note that this isn't a "free" operation even w/ fast-diff enabled. For each snapshot, it requires loading the object map for said snapshot from the OSDs to compute the usage. If this image metadata is cached / loading asynchronously in the background, it probably isn't too big of a deal. But if this will stall the UI while it's opening lots of images and computing usage, that might be sub-optimal.

Reply:
Thanks for the explanation. Yes, this code runs asynchronously and its results are cached.

Reply:
ah, you're right. I was trying to calculate something that only a filesystem stored inside image actually knows. Forget about that.